backend/protocol_rpc/endpoints.py

```diff
@@ -726,6 +726,101 @@ def get_contract(consensus_service: ConsensusService, contract_name: str) -> dic
     }
 
 
+def get_transactions_by_related_contract(
```

Comment:
@Henadek this method includes too many things that are not the responsibility of the endpoints layer. Here, we should only do validations and calls to the other services and processors.